uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -1112,12 +1112,98 @@ public function testFindOverlappingLeaveRequestsForOneOverlappingLeaveRequest()
     $startDate = '2016-11-01';
     $endDate = '2016-11-03';
 
-    $overlappingRequests = LeaveRequest::findOverlappingLeaveRequests($contactID, $startDate, $endDate);
+    $overlappingRequests = LeaveRequest::findOverlappingLeaveRequests($contactID, $startDate, $endDate, $this->absenceType->id, LeaveRequest::REQUEST_TYPE_LEAVE);
     $this->assertCount(1, $overlappingRequests);
     $this->assertInstanceOf(LeaveRequest::class, $overlappingRequests[0]);
     $this->assertEquals($leaveRequest1->id, $overlappingRequests[0]->id);
   }
 
+  public function testFindOverlappingLeaveRequestsForHoursUnitRequests() {
```

Comment:
I think there should actually be more detailed tests for this. Real case scenarios e.g A leave request exists 1pm to 2pm and 4pm to 5pm on a single day, would the staff be able to request leave for 3pm to 4:30pm which should not be possible.

Also tests for the inverse: would the staff be able to request leave for 2pm to 3pm ? which should be possible